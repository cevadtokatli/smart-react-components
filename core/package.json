{
  "name": "@smart-react-components/core",
  "version": "2.0.0-beta-0",
  "license": "MIT",
  "main": "index.js",
  "description": "SRC core includes the theme, Styled properties, and Element components.",
  "keywords": [
    "javascript",
    "typescript",
    "react",
    "styled-components"
  ],
  "author": {
    "name": "Cevad Tokatlı",
    "email": "cevadtokatli@hotmail.com"
  },
  "contributors": [
    {
      "name": "Cemil Tokatlı",
      "email": "cemil@tokatli.me"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/cevadtokatli/smart-react-components/tree/v2/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "yarn clear && tsc --outDir lib && yarn copy",
    "clear": "rm -rf lib",
    "copy": "cp package.json lib/package.json && cp README.md lib/README.md && cp ../LICENSE lib/LICENSE",
    "dev": "tsc -w --outDir lib",
    "lint": "eslint src",
    "release": "../scripts/release.sh core",
    "test": "jest"
  },
  "devDependencies": {
    "@types/color": "3.0.3",
    "@types/enzyme": "3.10.12",
    "@types/jest": "29.0.1",
    "@types/react": "18.0.19",
    "@types/react-dom": "18.0.6",
    "@types/styled-components": "5.1.26",
    "@wojtekmaj/enzyme-adapter-react-17": "0.6.7",
    "enzyme": "3.11.0",
    "eslint": "8.0.1",
    "jest": "29.0.3",
    "jsdom": "20.0.0",
    "jsdom-global": "3.0.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "styled-components": "5.3.5",
    "ts-jest": "29.0.0",
    "typescript": "4.8.3"
  },
  "dependencies": {
    "color": "4.2.3"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "styled-components": "^5.3.5"
  }
}
