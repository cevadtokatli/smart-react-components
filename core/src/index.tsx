export * from './theme'
export { default as createTheme } from './theme'
export * from './styled-props'
export * from './Element'
