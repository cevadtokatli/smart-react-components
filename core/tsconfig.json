{
  "extends": "../tsconfig.json",
  "include": ["src/**/*.tsx"],
  "exclude": ["node_modules", "src/**/*.test.tsx"]
}
