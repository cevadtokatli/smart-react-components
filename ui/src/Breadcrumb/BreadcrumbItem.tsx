import A from '@smart-react-components/core/Element/A'
import extractElementProps from '@smart-react-components/core/element-props'
import clickEvents, { ClickEvents } from '@smart-react-components/core/element-props/click-events'
import { Theme } from '@smart-react-components/core/theme'
import { ContentElement, JSXElementProps, PaletteProp, ShapeProp } from '@smart-react-components/core/types'
import React from 'react'
import { useTheme } from 'styled-components'
import BreadcrumbItemElement from '../components/Breadcrumb/BreadcrumbItemElement'
import { getWaveEffectPalette } from '../util/wave-effect'
import WaveEffect from '../WaveEffect'

export interface Props extends ClickEvents {
  children: ContentElement
  elementProps?: JSXElementProps
  href?: string
  isActive?: boolean
  isDisabled?: boolean
}

export interface PrivateProps {
  hasWaveEffect: boolean
  isOutline: boolean
  isSoft: boolean
  palette: PaletteProp
  separator: string
  shape: ShapeProp
  waveEffectPalette: PaletteProp
}

const BreadcrumbItem: React.FC<Props> = (props: Props & PrivateProps) => {
  const theme = useTheme() as Theme

  const waveEffectPalette = React.useMemo(() => getWaveEffectPalette(props, theme.$.vars.isDarkMode), [props.palette, props.isOutline, props.isSoft, props.waveEffectPalette, theme.$.vars.isDarkMode])

  const AnchorEl = (
    <A
      {...extractElementProps(props, [clickEvents])}
      {...props.elementProps}
    >
      {props.children}
    </A>
  )

  const El = props.hasWaveEffect ? <WaveEffect palette={waveEffectPalette}>{AnchorEl}</WaveEffect> : AnchorEl

  return (
    <BreadcrumbItemElement
      isActive={props.isActive}
      isDisabled={props.isDisabled}
      isOutline={props.isOutline}
      isSoft={props.isSoft}
      palette={props.palette}
      separator={props.separator}
      shape={props.shape}
    >
      {El}
    </BreadcrumbItemElement>
  )
}

export default BreadcrumbItem
