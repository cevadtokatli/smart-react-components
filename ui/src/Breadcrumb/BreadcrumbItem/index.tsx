import A from '@smart-react-components/core/Element/A'
import extractElementProps from '@smart-react-components/core/element-props'
import clickEvents, { ClickEvents } from '@smart-react-components/core/element-props/click-events'
import { ContentElement, JSXElementProps, PaletteProp, ShapeProp } from '@smart-react-components/core/types'
import React from 'react'
import BreadcrumbItemElement from './BreadcrumbItemElement'
import WaveEffect from '../../WaveEffect'

export interface Props extends ClickEvents {
  children: ContentElement
  elementProps?: JSXElementProps
  href?: string
  isActive?: boolean
  isDisabled?: boolean
}

export interface PrivateProps {
  hasWaveEffect: boolean
  isOutline: boolean
  isSoft: boolean
  palette: PaletteProp
  separator: string
  shape: ShapeProp
  waveEffectPalette: PaletteProp
}

const BreadcrumbItem: React.FC<Props> = (props: Props & PrivateProps) => {
  const AnchorEl = (
    <A
      {...extractElementProps(props, [clickEvents])}
      {...props.elementProps}
    >
      {props.children}
    </A>
  )
  const El = props.hasWaveEffect ? <WaveEffect palette={props.waveEffectPalette}>{AnchorEl}</WaveEffect> : AnchorEl

  return (
    <BreadcrumbItemElement
      isActive={props.isActive}
      isDisabled={props.isDisabled}
      isOutline={props.isOutline}
      isSoft={props.isSoft}
      palette={props.palette}
      separator={props.separator}
      shape={props.shape}
    >
      {El}
    </BreadcrumbItemElement>
  )
}

export default BreadcrumbItem
