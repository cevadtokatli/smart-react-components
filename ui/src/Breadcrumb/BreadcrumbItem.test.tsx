import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import BreadcrumbItem from './BreadcrumbItem'

describe('<BreadcrumbItem />', () => {
  let props

  beforeAll(() => [
    props = {
      hasWaveEffect: true,
      isOutline: false,
      isSoft: false,
      palette: 'primary',
      separator: null,
      shape: 'rectangle',
      waveEffectPalette: 'light',
    }
  ])

  it('should render component', () => {
    const { asFragment } = render(wrapTheme(
      <BreadcrumbItem {...props}>Breadcrumb Item</BreadcrumbItem>
    ))
    expect(asFragment()).toMatchSnapshot()
  })
})
