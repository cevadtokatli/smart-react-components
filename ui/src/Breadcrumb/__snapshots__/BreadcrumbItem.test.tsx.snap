// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadcrumbItem /> should render component 1`] = `
<DocumentFragment>
  .c1 {
  isolation: isolate;
  overflow: hidden;
  position: relative;
  -webkit-tap-highlight-color: transparent;
}

.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 0;
}

.c0 > a {
  border-radius: .125rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: .475rem .6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 200ms 0s ease-in-out;
  transition: 200ms 0s ease-in-out;
  -webkit-transition-property: background,border,color,fill;
  transition-property: background,border,color,fill;
  color: #fff;
  fill: #fff;
}

.c0 > a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:not(:last-child)::after {
  content: "/";
  display: inline-block;
  padding: 0 1px;
  color: #fff;
}

<li
    class=" c0"
    shape="rectangle"
  >
    <a
      class="c1"
    >
      Breadcrumb Item
    </a>
  </li>
</DocumentFragment>
`;
